pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicsAuthTest.java

```diff
@@ -0,0 +1,236 @@
+        conf.setAuthenticationProviders(providers);
+        super.internalSetup();
+        PulsarAdminBuilder pulsarAdminBuilder = PulsarAdmin.builder().serviceHttpUrl(brokerUrl != null
+                ? brokerUrl.toString() : brokerUrlTls.toString())
+                .authentication(AuthenticationToken.class.getName(),
+                        ADMIN_TOKEN);
+        admin = Mockito.spy(pulsarAdminBuilder.build());
+        admin.clusters().createCluster(testLocalCluster, new ClusterDataImpl());
+        admin.tenants().createTenant(testTenant, new TenantInfoImpl(Sets.newHashSet("role1", "role2"),
+                Sets.newHashSet(testLocalCluster)));
+        admin.namespaces().createNamespace(testTenant + "/" + testNamespace,
+                Sets.newHashSet(testLocalCluster));
+        admin.namespaces().grantPermissionOnNamespace(testTenant + "/" + testNamespace, "producer",
+                EnumSet.of(AuthAction.produce));
+        admin.namespaces().grantPermissionOnNamespace(testTenant + "/" + testNamespace, "consumer",
+                EnumSet.of(AuthAction.consume));
+    }
+
+    @Override
+    @AfterMethod
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    @DataProvider(name = "variations")
+    public static Object[][] variations() {
+        return new Object[][]{
+                {CONSUME_TOKEN, 401},
+                {PRODUCE_TOKEN, 200}
+        };
+    }
+
+    @Test(dataProvider = "variations")
+    public void testProduceToNonPartitionedTopic(String token, int status) throws Exception {
+        admin.topics().createNonPartitionedTopic("persistent://" + testTenant + "/"
+                + testNamespace + "/" + testTopicName);
+        Schema<String> schema = StringSchema.utf8();
+        ProducerMessages producerMessages = new ProducerMessages();
+        producerMessages.setKeySchema(ObjectMapperFactory.getThreadLocal().
+                writeValueAsString(schema.getSchemaInfo()));
+        producerMessages.setValueSchema(ObjectMapperFactory.getThreadLocal().
+                writeValueAsString(schema.getSchemaInfo()));
+        String message = "[" +
+                "{\"key\":\"my-key\",\"payload\":\"RestProducer:1\",\"eventTime\":1603045262772,\"sequenceId\":1}," +
+                "{\"key\":\"my-key\",\"payload\":\"RestProducer:2\",\"eventTime\":1603045262772,\"sequenceId\":2}]";
+        producerMessages.setMessages(ObjectMapperFactory.getThreadLocal().readValue(message,
+                new TypeReference<List<ProducerMessage>>() {
+                }));
+
+        WebTarget root = buildWebClient();
+        Response response = root.path("/topics/persistent/" + testTenant + "/" + testNamespace + "/"
+                + testTopicName).request(MediaType.APPLICATION_JSON)
+                .header("Authorization", "Bearer " + token)
+                .post(Entity.json(producerMessages));
+        Assert.assertEquals(response.getStatus(), status);
```

Comment:
There are some repeated code blocks, maybe we could try to reuse the same code block. It seems that only topic names are different.